src/libstd/thread/local.rs

```diff
@@ -106,12 +106,27 @@ pub struct LocalKey<T: 'static> {
 #[stable(feature = "rust1", since = "1.0.0")]
 #[allow_internal_unstable]
 macro_rules! thread_local {
-    (static $name:ident: $t:ty = $init:expr) => (
-        static $name: $crate::thread::LocalKey<$t> =
+    // rule 1: process multiple declarations where the first one is private
+    ($(#[$attr:meta])* static $name:ident: $t:ty = $init:expr; $($rest:tt)+) => (
+        thread_local!($(#[$attr])* static $name: $t = $init); // go to rule 2
+        thread_local!($($rest)*);
+    );
+
+    // rule 2: handle a single private declaration
+    ($(#[$attr:meta])* static $name:ident: $t:ty = $init:expr $(;)*) => (
```

Comment:
This'll unfortunately accept syntax like:

```rust
thread_local!(static FOO: usize = 2 ;;;;;;;;;;;;;;;;;;;;;);
```

Although I'm not sure how much we can do about that.